runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -402,34 +401,26 @@ public CorfuRuntime registerBeforeRpcHandler(Runnable handler) {
      * a router.
      */
     @Getter
-    @Setter
-    public Function<String, IClientRouter> getRouterFunction = overrideGetRouterFunction != null
-            ? (address) -> overrideGetRouterFunction.apply(this, address) : (address) ->
-                nodeRouters.compute(address, (k, r) -> {
-                    final NettyClientRouter router = (NettyClientRouter) r;
-                    if (router != null) {
-                        // Return an existing router if we already have one and it is connected.
-                        return router;
-                    } else {
-                        NodeLocator node = NodeLocator.parseString(address);
-                        // Generate a new router, start it and add it to the table.
-                        NettyClientRouter newRouter = new NettyClientRouter(node,
-                            getNettyEventLoop(),
-                            getParameters());
-                        log.debug("Connecting to new router {}", node);
-                        try {
-                            newRouter.addClient(new LayoutClient())
-                                .addClient(new SequencerClient())
-                                .addClient(new LogUnitClient().setMetricRegistry(metrics != null
-                                    ? metrics : CorfuRuntime.getDefaultMetrics()))
-                                .addClient(new ManagementClient());
-                        } catch (Exception e) {
-                            log.warn("Error connecting to router", e);
-                            throw e;
-                        }
-                        return newRouter;
-                    }
-                });
+    private final Function<String, IClientRouter> getRouterFunction =
+            overrideGetRouterFunction != null ? (address) ->
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "address" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)